solo/solo.py

```diff
@@ -165,10 +165,16 @@ def main():
         'ignore_batch', False) else scvi_data.n_batches
 
     # training parameters
+    batch_size = params.get('batch_size', 128)
     valid_pct = params.get('valid_pct', 0.1)
     learning_rate = params.get('learning_rate', 1e-3)
     stopping_params = {'patience': params.get('patience', 10), 'threshold': 0}
 
+    # protect against single example batch
+    while num_cells % batch_size == 1:
+        batch_size = int(np.round(1.25*batch_size))
```

Comment:
What would be wrong with?:
```
batch_size = int(np.round(1 + batch_size))
```
Also wouldn't need to put this in a `while` loop then.

Reply:
Yea that's probably fine. GPUs tend to achieve better efficiency with batches divisible by powers of two, so I tried to come up with a strategy that would tend to maintain that. Either way is fine with me.

Reply:
Gotcha. I figured there might be something like that. This is fine with me. Merge away!